flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectResultFetcher.java

```diff
@@ -125,7 +127,13 @@ public T next() throws IOException {
                 try {
                     response = sendRequest(buffer.getVersion(), requestOffset);
                 } catch (Exception e) {
-                    LOG.warn("An exception occurs when fetching query results", e);
+                    if (ExceptionUtils.findThrowable(
+                                    e, UnavailableDispatcherOperationException.class)
```

Comment:
Assuming that `collect()` is mostly used for prototyping we should cover all cases with this change (because leader election is quick).
It would be nice if one could differentiate between timeouts at the various layers (Dispatcher, JobManager, actual coordinator operation), but that's not something we can fix easily :/

Reply:
True. Let's not do it right now.